yapapi/executor/services.py

```diff
@@ -60,52 +59,65 @@ class ServiceState(statemachine.StateMachine):
 
 @dataclass
 class ServiceSignal:
-    """
-    Simple container to carry information between the client code and the Service instance.
-    """
+    """Simple container to carry information between the client code and the Service instance."""
 
     message: Any
     response_to: Optional["ServiceSignal"] = None
 
 
 class Service:
-    """
-    Base Service class to be extended by application developers to define their own,
-    specialized Service specifications.
+    """Base class for service specifications.
+
+    To be extended by application developers to define their own, specialized
+    Service specifications.
     """
 
     def __init__(self, cluster: "Cluster", ctx: WorkContext):
+        """Initialize a service for a specific cluster of instances and a specific work context.
+
+        :param cluster: a cluster to which an instance of this service belongs
+        :param ctx: a work context object for executing commands on a provider that runs this
+            service instance.
+        """
         self._cluster: "Cluster" = cluster
         self._ctx: WorkContext = ctx
 
         self.__inqueue: asyncio.Queue[ServiceSignal] = asyncio.Queue()
         self.__outqueue: asyncio.Queue[ServiceSignal] = asyncio.Queue()
-        self.post_init()
-
-    def post_init(self):
-        pass
 
     @property
     def id(self):
+        """Return the id of the work context associated with this service instance."""
```

Comment:
well, I'm unsure - my thought here was that it would be "an identifier of an instance" and it would an implementation detail that it's - effectively - equal to the id of the respective work context ...

Reply:
fixed in 761f820